pkg/execution/debounce/lua/updateDebounce.lua

```diff
@@ -56,12 +55,15 @@ end
 -- Check that the queue item is not leased (ie. this debounce is not in progress)
 local item = get_queue_item(keyQueueHash, queueJobID)
 if item == nil then
-	-- The queue item was not found.  Return a new debounce.
-	return -3
+	-- The queue item was not found. return not found but set the debounce in the hash map
+  -- for lookup
+  redis.call("SETEX", keyPtr, ttl, debounceID)
+  redis.call("HSET", keyDbc, debounceID, debounce)
```

Comment:
make sure to create a new debounce, update the TTL and queue it if it's not found.